### ISSUE TITLE
`indices` behaves differently when dealing with arrays vs strings

### ISSUE THREAD

**Issue by kklingenberg** [jqlang/jq#3030]
**Describe the bug**

Not sure if it is a bug. The behaviour of `indices` differs when given arrays vs strings when the matches overlap.

**To Reproduce**

```jq
[0, 0, 0] | indices([0, 0]) == [0, 1]
```

vs.

```jq
"aaa" | indices("aa") == [0]
```

**Expected behavior**

Both cases should match, or if the difference is intented the string case should be documented.

**Environment (please complete the following information):**

- OS and Version: Ubuntu 22.04
- jq version 1.7

**Additional context**

Related issue in jaq: <https://github.com/01mf02/jaq/issues/154>

**Comment by itchyny** [comment]
Previously jq 1.6 worked like this but I think jq 1.7 works as expected (fixed by #2718).

**Comment by kklingenberg** [comment]
You're right, I accidentally used the previous version of jq when running the example :(

Sorry.
